### DISCUSSION TITLE
关于tlshelper模块的错误异常的处理

### DISCUSSION THREAD

**Discussion by skynetlua** [cloudwu/skynet#1072] [Q&A]
tlshelper模块经常会因为各种问题抛出异常，能不能对这些错误异常进行处理，返回相应的http错误代码，以便把这个错误代码告诉客户机的浏览器。

**Comment by lvzixun** [comment]
你是再说`socket error`这个异常错误吗？

**Comment by skynetlua** [comment]
是的，socket错误和handshake错误异常都可能发生，可以加你qq么？

**Comment by lvzixun** [comment]
如果是`socket error`的话， 这个一般是网络问题， 可以把`internal.lua`抛出异常的地方加上详细的错误信息。 如果是`ssh`那边抛出来的错误， 确认下是否只有`ltls`这一个lib来调用`openssl`的接口。
